src/user_interface/keyboard.rs

```diff
@@ -110,8 +110,9 @@ impl Keyboard {
                     return None;
                 }
                 if action.is_some() {
-                    current.clear();
-                    current.push(self.menu.clone());
+                    if current.len() > 1 {
+                        current.pop();
+                    }
```

Comment:
```suggestion
                    current.pop();
```
Will just return `None` if current is empty, and won't panic